apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/data_updater/impl.ex

```diff
@@ -1466,15 +1456,10 @@ defmodule Astarte.DataUpdaterPlant.DataUpdater.Impl do
            interface_major: major,
            match_path: "/*"
          }} ->
-          with {:ok, db_client} <-
-                 Database.connect(realm: state.realm),
-               {:ok, %InterfaceDescriptor{aggregation: :individual}} <-
-                 InterfaceQueries.fetch_interface_descriptor(db_client, interface_name, major) do
+          with {:ok, db_client} <- Database.connect(realm: state.realm),
+               :ok <- InterfaceQueries.check_if_interface_exists(db_client, interface_name, major) do
```

Comment:
Let's check what `check_if_interface_exists` returns in case of missing interface, and make sure that it is properly managed by our "with else" block.

Reply:
`check_if_interface_exists` returns one of these errors:
```
      [count: 0] ->
        {:error, :interface_not_found}

      %{acc: _, msg: error_message} ->
        Logger.warn("check_if_interface_exists: database error: #{error_message}")
        {:error, :database_error}

      {:error, reason} ->
        Logger.warn("check_if_interface_exists: failed, reason: #{inspect(reason)}.")
        {:error, :database_error}
```

The call stack is:
```
(incoming AMQP RPC)
-> RPC.Handler.call_rpc/1
-> DataUpdater.handle_install_volatile_trigger/8
-> Server.handle_call/4
-> Impl.handle_install_volatile_trigger/7
```

The result is passed out as-is up the stack until it reaches `call_rpc` where it's matched like this:
```elixir
    with :ok <-
           DataUpdater.handle_install_volatile_trigger(
             realm_name,
             device_id,
             object_id,
             object_type,
             parent_id,
             simple_trigger_id,
             simple_trigger,
             trigger_target
           ) do
      %GenericOkReply{}
      |> encode_reply()
      |> ok_wrap()
    else
      {:error, reason} ->
        generic_error(reason)
    end
```